Mithril update 2023-12-13 (#307)
input-output-hk/cardano-updates: blog/2023-12-13-mithril.md

---
title: Mithril Team Update
slug: 2023-12-13-mithril
authors: jpraynaud
tags: [mithril]
hide_table_of_contents: false
---

## High level overview

> This week, the Mithril team finished adapting the Mithril client library for WASM compilation and providing a JavaScript API for it. They completed the proof-of-concept for proving that a subset of Cardano transactions is part of the whole transactions set. They also completed the refactoring of the Mithril client CLI with the Mithril client library. Additionally, the team kept working on the threat modeling and risk analysis for P2P networking and on the implementation of a testing Mithril network for SanchoNet.

Reviewer: ```suggestion
This week, the Mithril team completed the adaptation of the Mithril client library for WASM compilation, providing a JavaScript API for it. They also completed the proof of concept to demonstrate that a subset of Cardano transactions is part of the entire transactions set. The refactoring of the Mithril client CLI with the Mithril client library was also finalized. Additionally, the team kept working on the threat modeling and risk analysis for P2P networking, along with progressing in the implementation of a testing Mithril network for SanchoNet.
```

> Finally, they started implementing an expiration mechanism for the open messages in the aggregator, and they completed the upgrade of Rust dependencies with breaking changes.

Reviewer: ```suggestion
Finally, they started implementing an expiration mechanism for the open messages in the aggregator and completed the upgrade of Rust dependencies with breaking changes.
```

## Low level overview
- Completed the issue **Light Wallet: Implement `mithril-client` WASM library** [#1336](https://github.com/input-output-hk/mithril/issues/1336)

> - Completed the issue **Light Wallet: Certify Cardano UTxO/transactions set PoC** [#1393](https://github.com/input-output-hk/mithril/issues/1393)

Reviewer: ```suggestion
- Completed the issue **Light Wallet: Certify Cardano UTXO/transactions set PoC** [#1393](https://github.com/input-output-hk/mithril/issues/1393)
```

- Completed the issue **Refactor `mithril-client-cli` with `mithril-client` library** [#1335](https://github.com/input-output-hk/mithril/issues/1335)
- Worked on the issue **Implement `open_message` timeout in aggregator** [#1387](https://github.com/input-output-hk/mithril/issues/1387)
- Worked on the issue **P2P threat modeling and risk analysis** [#1350](https://github.com/input-output-hk/mithril/issues/1350)
- Worked on the issue **Enhance machine readable logs in client** [#1234](https://github.com/input-output-hk/mithril/issues/1234)

> - Worked on the issue **Create a test network on `Sanchonet`** [#1173](https://github.com/input-output-hk/mithril/issues/1173)

Reviewer: ```suggestion
- Worked on the issue **Create a test network on `SanchoNet`** [#1173](https://github.com/input-output-hk/mithril/issues/1173)
```
